packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1135,20 +1155,28 @@ List<PluginInterfaceResolution> resolvePlatformImplementation(
 
   for (final Plugin plugin in plugins) {
     for (final String platform in platforms) {
-      // The plugin doesn't implement this platform.
       if (plugin.platforms[platform] == null &&
           plugin.defaultPackagePlatforms[platform] == null) {
+        // The plugin doesn't implement this platform.
         continue;
       }
-      // The plugin doesn't implement an interface, verify that it has a default implementation.
-      final String? implementsPackage = plugin.implementsPackage;
+      String? implementsPackage = plugin.implementsPackage;
       if (implementsPackage == null || implementsPackage.isEmpty) {
         final String? defaultImplementation = plugin.defaultPackagePlatforms[platform];
-        if (defaultImplementation == null) {
+        final bool hasInlineDartImplementation =
+          plugin.pluginDartClassPlatforms[platform] != null;
+        if (defaultImplementation == null && !hasInlineDartImplementation) {
           if (throwOnPluginPubspecError) {
```

Comment:
fyi: `throwOnPluginPubspecError` is `false` due to backward compatibilities.
https://github.com/flutter/flutter/blob/4c6a9dcb404bbfe5afafe85ac8ac5b09cd3451c3/packages/flutter_tools/lib/src/build_system/targets/dart_plugin_registrant.dart#L58

Should it still print the error, but not exit with code?

Reply:
Since the goal is to eliminate this error in a follow-up PR, I think we can leave it as-is for now.